### PR TITLE
Update parent POM and solve requireUpperBoundDeps

### PR DESCRIPTION
**Pull request by casz**
* Answer the age old question
* Token macro uses newer work-flow-step
* Powermock uses a newer mockito
* Cleanup in POM

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,9 +1,10 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>2.10</version><!-- which version of Jenkins is this plugin built against? -->
+        <version>2.36</version><!-- Built against the Jenkins version specified in properties as 'jenkins.version' -->
```
Comment (casz) [inline pom.xml]: Updated Parent POM [changelog]( https://github.com/jenkinsci/plugin-pom/blob/master/CHANGELOG.md)

Comment noted here: https://github.com/jenkinsci/office-365-connector-plugin/blob/d27e25fca4c3a623f6f7fcbd8bd9b5a0d8007152/pom.xml#L27

File: pom.xml [hunk 2]
```diff
@@ -37,7 +38,7 @@
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-all</artifactId>
-            <version>1.9.5</version>
+            <version>1.10.8</version>
```
Comment (casz) [inline pom.xml]:
```
Require upper bound dependencies error for org.mockito:mockito-all:1.9.5 paths to dependency are:
+-org.jenkins-ci.plugins:Office-365-Connector:3.1-SNAPSHOT
  +-org.mockito:mockito-all:1.9.5
and
+-org.jenkins-ci.plugins:Office-365-Connector:3.1-SNAPSHOT
  +-org.powermock:powermock-api-mockito:1.6.1
    +-org.mockito:mockito-all:1.10.8
```

File: pom.xml [hunk 3]
```diff
@@ -84,7 +79,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-step-api</artifactId>
-            <version>1.11</version>
+            <version>2.7</version>
```
Comment (casz) [inline pom.xml]:
```
Require upper bound dependencies error for org.jenkins-ci.plugins.workflow:workflow-step-api:1.11 paths to dependency are:
+-org.jenkins-ci.plugins:Office-365-Connector:3.1-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-step-api:1.11
and
+-org.jenkins-ci.plugins:Office-365-Connector:3.1-SNAPSHOT
  +-org.jenkins-ci.plugins:token-macro:2.3
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.7
and
+-org.jenkins-ci.plugins:Office-365-Connector:3.1-SNAPSHOT
  +-org.jenkins-ci.plugins:token-macro:2.3
    +-org.jenkins-ci.plugins.workflow:workflow-job:2.0
      +-org.jenkins-ci.plugins.workflow:workflow-support:1.15
        +-org.jenkins-ci.plugins.workflow:workflow-step-api:1.15
and
+-org.jenkins-ci.plugins:Office-365-Connector:3.1-SNAPSHOT
  +-org.jenkins-ci.plugins:token-macro:2.3
    +-org.jenkins-ci.plugins.workflow:workflow-job:2.0
      +-org.jenkins-ci.plugins.workflow:workflow-support:1.15
        +-org.jenkins-ci.plugins.workflow:workflow-api:1.15
          +-org.jenkins-ci.plugins.workflow:workflow-step-api:1.15
```

File: pom.xml [hunk 4]
```diff
@@ -70,12 +71,6 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.jenkins-ci.main</groupId>
-            <artifactId>jenkins-war</artifactId>
```
Comment (casz) [inline pom.xml]: Provided by Parent POM

File: pom.xml (hunk 3 above)
Comment (casz) [reply pom.xml]: With #24 we also follow that api 👍

File: pom.xml (hunk 1 above)
Comment (SrhebbarMicrosoft) [reply pom.xml]: @casz Change in the version from 2.10 to 2.36 is failing in Netbeans. Am getting the below error:
--- maven-enforcer-plugin:3.0.0-M1:display-info (display-info) @ Office-365-Connector ---
Maven Version: 3.0.5
JDK Version: 1.8.0_144 normalized as: 1.8.0-144
OS Info: Arch: amd64 Family: windows Name: windows 10 Version: 10.0

--- maven-enforcer-plugin:3.0.0-M1:enforce (display-info) @ Office-365-Connector ---
Ignoring requireUpperBoundDeps in com.google.guava:guava
Ignoring requireUpperBoundDeps in com.google.code.findbugs:jsr305
Rule 0: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
3.1.0 required by frontend-maven-plugin at least.

My maven version is 3.3.9... Am not sure why netbeans is not taking the higher version. My IDE is up to date. Any Idea how I can fix this?

Comment (SrhebbarMicrosoft) [reply pom.xml]: It's the built in I guess... That is the reason it is failing. When I build it externally, everything works as the version of maven installed in 3.3.9... How can I change it in netbeans?

Comment (casz) [reply pom.xml]: Tools > Options > Java > Maven > Maven Home...
image

Comment (SrhebbarMicrosoft) [reply pom.xml]: Thanks a ton... :)

**Pull request merged**